File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/29/2026, 10:14:17 AM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):  # Avoid checking pairs twice
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []  # Return an empty list if no solution is found

File: 1-TwoSum/1-TwoSum.rs
// Last updated: 7/29/2026, 10:14:14 AM
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut number_map: HashMap<i32, i32> = HashMap::new();
        for (i, &num) in nums.iter().enumerate() {
            let complement = target - num;
            if let Some(&index) = number_map.get(&complement) {
                return vec![index, i as i32];
            }
            number_map.insert(num, i as i32);
        }
        vec![]  // Return an empty vector if no solution is found
    }
}